### DISCUSSION TITLE
modificare sonoff

### DISCUSSION THREAD

**Discussion by 72tommy** [indomus/forum#2655] [Hardware & Firmware]
buongiorno a tutti, sono nuovo del forum, sono qui perchè ho in mente un progetto e vorrei imparare come realizzarlo; come da titolo, vorrei modificare il funzionamento di un sonoff (un dual R3) proprio nelle sue funzioni, vorrei sapere se è possibile copiare il firmware che è attualmente installato nel dispositivo e successivamente, dopo averlo modificato, copiarlo di nuovo nel sonoff; non mi interessano le funzioni smart, che funzioni o meno con l'app ewelink, a me interessano le funzioni del relè ma come le voglio io, poichè in commercio non esiste nulla che faccia al caso mio, perciò mi è venuta questa idea; avevo già acquistao tempo addietro l'adattatore FTDI232, inoltre sono un appassionato di elettronica e un pò smanettone per cui non mi spaventa cimentarmi in cose nuove. Grazie a tutti quelli che mi potranno dare una mano.

**Comment by essegi65** [comment]
Non so se riuscirai a modificare il FW originale, però forse potresti vedere come funziona Tasmota che è un altro FW x sonoff che puoi installare. Credi anche ESPHOME funzioni con sonoff e li puoi personalizzarlo come vuoi

**Comment by indomus** [comment]
> buongiorno a tutti, sono nuovo del forum, sono qui perchè ho in mente un progetto e vorrei imparare come realizzarlo; come da titolo, vorrei modificare il funzionamento di un sonoff (un dual R3) proprio nelle sue funzioni, vorrei sapere se è possibile copiare il firmware che è attualmente installato nel dispositivo e successivamente, dopo averlo modificato, copiarlo di nuovo nel sonoff; non mi interessano le funzioni smart, che funzioni o meno con l'app ewelink, a me interessano le funzioni del relè ma come le voglio io, poichè in commercio non esiste nulla che faccia al caso mio, perciò mi è venuta questa idea; avevo già acquistao tempo addietro l'adattatore FTDI232, inoltre sono un appassionato di elettronica e un pò smanettone per cui non mi spaventa cimentarmi in cose nuove. Grazie a tutti quelli che mi potranno dare una mano.

https://indomus.it/guide/backup-restore-e-cancellazione-del-firmware-di-un-sonoff-e-simili/
